torch/_export/serialize.py

```diff
@@ -46,33 +69,411 @@ def _reverse_map(d):
+    ret = {}
+    if stack_trace := node.meta.get("stack_trace"):
+        ret["stack_trace"] = stack_trace
+    module_fqn = node.meta.get("module_fqn")
+    # Need an explicit None check instead of walrus operator, because
+    # module_fqn can be the empty string if the node belongs to the root.
+    # The walrus operator returns False on an empty string :(
+    if module_fqn is not None:
+        ret["module_fqn"] = module_fqn
+    # TODO(angelayi) add nn_module_stack and source_fn
+    return ret
+
+
+def export_operator(target) -> str:
+    if isinstance(target, str):
+        return target
+    elif target in _SYM_INT_OPS:
+        return f"{target.__module__}.{target.__name__}"
+    elif isinstance(target, torch._ops.HigherOrderOperator):
+        return target.__name__
+    else:
+        return str(target)
+
+
+def export_call_spec(call_spec: ep.CallSpec) -> CallSpec:
+    # TODO(angelayi): spec
+    return CallSpec(in_spec="", out_spec="")
+
+
+def export_signature(sig: ep.ExportGraphSignature) -> GraphSignature:
+    if bw_sig := sig.backward_signature:
+        backward_signature = BackwardSignature(
+            gradients_to_parameters=bw_sig.gradients_to_parameters,
+            gradients_to_user_inputs=bw_sig.gradients_to_user_inputs,
+            loss_output=bw_sig.loss_output,
+        )
+    else:
+        backward_signature = None
+
+    graph_signature = GraphSignature(
+        inputs_to_parameters=sig.inputs_to_parameters,
+        inputs_to_buffers=sig.inputs_to_buffers,
+        user_inputs=sig.user_inputs,
+        user_outputs=sig.user_outputs,
+        buffers_to_mutate=sig.buffers_to_mutate,
+        backward_signature=backward_signature,
+    )
+    return graph_signature
+
+
+def export_state_dict(state_dict: Dict[str, Any]) -> bytes:
+    buffer = io.BytesIO()
+    state_dict = dict(state_dict)
+    for name in state_dict:
+        # This is a workaround for backend's tensor deserialization problem:
+        # unpickleTensor() always create a tensor on the device where it was originally saved
+        # This behavior is bad for multi-gpu training, as we wish to directly load the tensor
+        # on the designated device.
+        # For now, we simply move the tensor to cpu before saving.
+        # TODO: this should be fixed by deserialization instead.
+        state_dict[name] = state_dict[name].cpu()
+    torch.save(state_dict, buffer)
+    return buffer.getvalue()
+
+
+def _is_single_tensor_return(target: torch._ops.OpOverload) -> bool:
+    returns = target._schema.returns
+    return len(returns) == 1 and isinstance(returns[0].real_type, torch.TensorType)
+
+
+class Serializer:
+    def __init__(self, opset_version: int = 0):
+        self.inputs: List[Argument] = []
+        self.outputs: List[Argument] = []
+        self.nodes: List[Node] = []
+        self.tensor_values: Dict[str, TensorValue] = {}
+        self.sym_int_values: Dict[str, SymInt] = {}
+        self.opset_version: int = opset_version
```

Comment:
Is it possible to do a `Dict[str, Int]` for opset_version? I think it's clear that we want to serialize an opset version per namespace. cc @avikchaudhuri

Reply:
yeah I think we should pass in a dict.